StRoot/StEvent/StETofHeader.cxx

```diff
@@ -72,12 +78,25 @@ StETofHeader::StETofHeader( const double& trgGdpbTime, const double& trgStarTime
 {
     setRocGdpbTs( gdpbTs );
     setRocStarTs( starTs );
-    //const size_t kNbGet4sInSystem = 1728;
-    //mMissMatchFlagVec.resize( kNbGet4sInSystem );
-    //for( auto mapcheck : mMissMatchFlagVec ){
-    //	cout << mapcheck << endl;
-    // }
+    const size_t kNbCountersInSystem = 108;
+    mGoodEventFlagVec = std::vector<bool>( kNbCountersInSystem, false );      
+}
 
+StETofHeader::StETofHeader( const double& trgGdpbTime, const double& trgStarTime,
+                            const map< unsigned int, uint64_t >& gdpbTs, const map< unsigned int, uint64_t >& starTs,
+                            const unsigned int& starToken, const unsigned int& starDaqCmdIn, const unsigned int& starTrgCmdIn,
+                            const uint64_t& eventStatusFlag, const std::vector<bool>& MissMatchFlagVec, const std::vector<bool>& GoodEventFlagVec  )
+: mTrgGdpbFullTime( trgGdpbTime ),
+  mTrgStarFullTime( trgStarTime ),
+  mStarToken( starToken ),
+  mStarDaqCmdIn( starDaqCmdIn ),
+  mStarTrgCmdIn( starTrgCmdIn ),
+  mEventStatusFlag( eventStatusFlag ),
+  mMissMatchFlagVec( MissMatchFlagVec ),
+  mGoodEventFlagVec( GoodEventFlagVec )
+{
+    setRocGdpbTs( gdpbTs );
+    setRocStarTs( starTs ); 
```

Comment:
You could use delegating constructors to reduce all of this boireplate code for StETofHeader constructors.